esp-hal/src/psram/esp32s3.rs

```diff
@@ -60,12 +60,40 @@ pub(crate) fn init_psram(config: PsramConfig) {
     let mut config = config;
     utils::psram_init(&mut config);
 
+    #[cfg(instruction_cache_size_32kb)]
+    const CONFIG_ESP32S3_INSTRUCTION_CACHE_SIZE: u32 = 0x8000;
+    #[cfg(instruction_cache_size_16kb)]
     const CONFIG_ESP32S3_INSTRUCTION_CACHE_SIZE: u32 = 0x4000;
+
+    #[cfg(icache_associated_ways_8)]
```

Comment:
Could you turn these into `if cfg!()`? this is very visually busy

Reply:
Agreed. Was trying to recall the syntax but failed. I knew you would see this and remind me :joy: